Modules/FireForget/Get-AbortCleanKansaServersFF.ps1

```diff
@@ -0,0 +1,184 @@
+# This module is used to abort the deployment of a Kansa module from distributed-kansa servers
+# It can also be used in the event that a kansa server stalls its deployment due to too many connection
+# errors or timeouts.  The parameters can be configured to clean up result folders and error logs
+# and optionally send any uncollected results/errors to ELK for the previous n kansa jobs.
+# The default settings are set for safe operation and will only send results/errors from the last 1 job
+
+if(!(Get-Variable -Name KillRunningKansaJob -ErrorAction SilentlyContinue)){$KillRunningKansaJob = $False}
+if(!(Get-Variable -Name CleanupServer -ErrorAction SilentlyContinue)){$CleanupServer = $False}
+if(!(Get-Variable -Name SendPartialResultsToElk -ErrorAction SilentlyContinue)){$SendPartialResultsToElk = $True}
+if(!(Get-Variable -Name DeletePartialResults -ErrorAction SilentlyContinue)){$DeletePartialResults = $False}
+if(!(Get-Variable -Name NumRecentJobs -ErrorAction SilentlyContinue)){$NumRecentJobs=1}
+if(!(Get-Variable -Name KansaRemotePath -ErrorAction SilentlyContinue)){$KansaRemotePath="F:\Kansa"}
```

Comment:
Why the hardcoded F:\? Is there not a more flexible way to do this? What if someone runs this an has an F:\ assigned for something else?

Reply:
Great question.  We left the default path as "F:\Kansa" but the Fire&Forget modules are designed to have all of these variables passed-in as realtime parameters at launch.  That's why the line checks to see if the variable is already defined before it sets it to the default value.  If it would make more sense, we can set the default value for KansaRemotePath to "" or $null.  But the intent of this block is to let the analyst know which runtime variables need to be dynamically passed-in/defined when the module is launched

Reply:
Got it, seems reasonable as is.